src-java/kilda-pce/src/main/java/org/openkilda/pce/impl/InMemoryPathComputer.java

```diff
@@ -114,57 +122,144 @@ private GetPathsResult getPath(AvailableNetwork network, Flow flow,
+            throws UnroutableFlowException, RecoverableException {
+        AvailableNetwork network = availableNetworkFactory.getAvailableNetwork(haFlow, new ArrayList<>());
+
+        if (haFlow.getSubFlows() == null || haFlow.getSubFlows().size() != 2) {
+            throw new IllegalArgumentException(format("HA-flow %s must have 2 sub flows to find a path, but it "
+                    + "has following sub flows: %s", haFlow.getHaFlowId(), haFlow.getSubFlows()));
+        }
+
+        List<GetPathsResult> paths = new ArrayList<>();
+        for (HaSubFlow subFlow : haFlow.getSubFlows()) {
+            paths.add(getPath(network, new RequestedPath(haFlow, subFlow), isProtected));
+        }
+
+        return GetHaPathsResult.builder()
+                .forward(unitePathsToHaPath(paths.get(0).getForward(), paths.get(1).getForward(), true))
+                .reverse(unitePathsToHaPath(paths.get(0).getReverse(), paths.get(1).getReverse(), false))
+                .backUpPathComputationWayUsed(
+                        paths.get(0).isBackUpPathComputationWayUsed() || paths.get(1).isBackUpPathComputationWayUsed())
+                .build();
+    }
+
+    private HaPath unitePathsToHaPath(Path firstPath, Path secondPath, boolean forward) {
+        HaPathBuilder haPath = HaPath.builder();
+        if (forward) {
+            if (!firstPath.getSrcSwitchId().equals(secondPath.getSrcSwitchId())) {
+                throw new IllegalArgumentException(format("Forward HA sub paths must have equal source switch."
+                        + "The first sub path: %s, the  second sub path: %s", firstPath, secondPath));
+            }
+            haPath.sharedSwitchId(firstPath.getSrcSwitchId());
+            haPath.yPointSwitchId(findYPointForForwardPaths(firstPath, secondPath));
+        } else {
+            if (!firstPath.getDestSwitchId().equals(secondPath.getDestSwitchId())) {
+                throw new IllegalArgumentException(format("Reverse HA sub paths must have equal destination switch."
+                        + "The first sub path: %s, the  second sub path: %s", firstPath, secondPath));
+            }
+            haPath.sharedSwitchId(firstPath.getDestSwitchId());
+            haPath.yPointSwitchId(findYPointForReversePaths(firstPath, secondPath));
+        }
+
+        haPath.subPaths(Lists.newArrayList(firstPath, secondPath));
+        haPath.latency(Math.max(firstPath.getLatency(), secondPath.getLatency()));
+        haPath.isBackupPath(firstPath.isBackupPath() || secondPath.isBackupPath());
+        haPath.minAvailableBandwidth(Math.min(
+                firstPath.getMinAvailableBandwidth(), secondPath.getMinAvailableBandwidth()));
+        return haPath.build();
+    }
+
+    private static SwitchId findYPointForForwardPaths(Path firstPath, Path secondPath) {
+        int id = findLastForwardSharedSegmentId(firstPath.getSegments(), secondPath.getSegments());
+        if (id == -1) {
+            return firstPath.getSrcSwitchId();
```

Comment:
In case there are no equal end points, this method is still say that a Y-point exists.

Reply:
good point. Fixed